test/test_testing.py

```diff
@@ -438,6 +438,23 @@ def test_assert_messages(self, device):
         self.assertEqual("no_user_msg", self._get_assert_msg(msg=None, debug_msg="no_user_msg"))
         self.assertEqual("debug_msg\nuser_msg", self._get_assert_msg(msg="user_msg", debug_msg="debug_msg"))
 
+    @onlyCUDA
+    @slowTest
+    def test_cuda_assert_should_stop_test_suite(self, device):
+        # This test is slow because it spawn another process to run another test suite.
+        import os
+        import subprocess
+        import sys
+        curdir = os.path.dirname(os.path.abspath(__file__))
+        test_path = os.path.join(curdir, 'scripts', 'test_with_deterministic_cuda_assert_failures.py')
+
+        # Test running of cuda assert test suite should early terminate.
+        p = subprocess.run([sys.executable, test_path], capture_output=True, timeout=120)
```

Comment:
Want to repeat a feedback from @ngimel on my PR approaching the same idea:
Instead of adding a new file, can we just embed the assert script in the same file, since it is pretty small?
Something like `p=subprocess.run([sys.executable, '-c', 'import torch;x=torch.rand(10, device="cuda");y = x[torch.tensor([25])].cpu()']